nifi-api/src/main/java/org/apache/nifi/registry/FileVariableRegistry.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.nifi.registry;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Map;
+
+
+public abstract class FileVariableRegistry extends MultiMapVariableRegistry {
+
+    public FileVariableRegistry() {
+        super();
+    }
+
+    public FileVariableRegistry(File... files){
+        super();
+        addVariables(files);
+    }
+
+    public FileVariableRegistry(Path... paths){
+        super();
+        addVariables(paths);
+    }
+
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    public void addVariables(File ...files){
+        if(files != null) {
+            for (final File file : files) {
+                try {
+                    registry.addMap(convertFile(file));
+                } catch (IOException iex) {
+                    throw new IllegalArgumentException("A file provided was invalid.", iex);
```

Comment:
wrapped it because of it's use in the constructor, wasn't too keen on throwing IOE from the constructor as well.  Also  as I side note I think I can remove references to public.